crmsh/bootstrap.py

```diff
@@ -104,6 +105,179 @@ def init_qdevice(self):
+        return self._sbd_service_flag
+
+    @sbd_service_flag.setter
+    def sbd_service_flag(self, value):
+        """
+        Set flag to enable sbd service on join process
+        """
+        self._sbd_service_flag = value
+
+    def sbd_init(self):
+        """
+        Function sbd_init includes these steps:
+        1. Check local environment
+        2. Get sbd device from options or interactive mode
+        3. Initialize sbd device
+        4. Write config file /etc/sysconfig/sbd
+        """
+        self._check_environment()
+        self._get_sbd_device()
+        if not self.sbd_device and not self.diskless_sbd:
+            return
+        self._initialize_sbd()
+        self._update_configuration()
+        # If process work through here, consider it's ready for enable service
+        self._sbd_service_flag = True
+
+    def configured_sbd_device(self):
+        """
+        Gets currently configured SBD device, i.e. what's in /etc/sysconfig/sbd
+        """
+        conf = utils.parse_sysconfig(SYSCONFIG_SBD)
+        return conf.get("SBD_DEVICE")
+
+    def manage_sbd_service(self):
+        if self._sbd_service_flag:
+            invoke("systemctl enable sbd.service")
+        else:
+            invoke("systemctl disable sbd.service")
+
+    def configure_sbd_resource(self):
+        if self.sbd_device and self.configured_sbd_device():
+            if not invoke("crm configure primitive stonith-sbd stonith:external/sbd pcmk_delay_max=30s"):
+                error("Can't create stonith-sbd primitive")
+            if not invoke("crm configure property stonith-enabled=true"):
+                error("Can't enable STONITH for SBD")
+        elif self.diskless_sbd:
+            if not invoke("crm configure property stonith-enabled=true stonith-watchdog-timeout=5s"):
+                error("Can't enable STONITH for diskless SBD")
+
+    def _get_sbd_device(self):
+        if self.sbd_device:
+            self.sbd_device = self._parse_sbd_device()
+            self._verify_sbd_device()
+        elif not self.diskless_sbd:
+            self._get_sbd_device_interactive()
+
+    def _initialize_sbd(self):
+        if self.diskless_sbd:
+            return
+        status_long('Initializing SBD...')
+        for dev in self.sbd_device:
+            if not invoke("sbd -d {} create".format(dev)):
+                error("Failed to initialize SBD device {}".format(dev))
+        status_done()
+
+    def _update_configuration(self):
+        self.sbd_config_dict["SBD_WATCHDOG_DEV"] = detect_watchdog_device()
+        if self.sbd_device:
+            self.sbd_config_dict["SBD_DEVICE"] = ';'.join(self.sbd_device)
+        utils.sysconfig_set(SYSCONFIG_SBD, **self.sbd_config_dict)
+        csync2_update(SYSCONFIG_SBD)
+
+    def _get_sbd_device_interactive(self):
+        if _context.yes_to_all:
+            warn("Not configuring SBD (%s left untouched)." % (SYSCONFIG_SBD))
+            return
+
+        status("""
```

Comment:
Put this string in a constant for this Class, it will be better

Reply:
Changed

Reply:
Changed